pkg/operator/ceph/object/objectstore.go

```diff
@@ -88,25 +106,185 @@ func deleteRealmAndPools(context *Context, spec cephv1.ObjectStoreSpec) error {
 	return nil
 }
 
-func setMultisite(context *Context, serviceIP string, spec cephv1.ObjectStoreSpec, realmName string, zoneGroupName string, zoneName string) error {
+func checkZoneIsMaster(context *Context, realmArg string, zoneGroupArg string, zoneArg string) (bool, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zoneGroupJson, err := DecodeZoneGroupConfig(zoneGroupOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse zonegroup get json")
+	}
+
+	zoneOutput, err := RunAdminCommandNoRealm(context, "zone", "get", realmArg, zoneGroupArg, zoneArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone")
+	}
+	zoneID, err := decodeID(zoneOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse zone id")
+	}
+
+	if zoneID == zoneGroupJson.MasterZoneID {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func checkZoneGroupIsMaster(context *Context, realmArg string, zoneGroupArg string) (bool, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zoneGroupJson, err := DecodeZoneGroupConfig(zoneGroupOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse master zone id")
+	}
+
+	zoneGroupIsMaster, err := strconv.ParseBool(zoneGroupJson.IsMaster)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse is_master from zone group json into bool")
+	}
+
+	return zoneGroupIsMaster, nil
+}
+
+func DecodeSecret(secret *v1.Secret, keyName string) (string, error) {
+	realmKey, ok := secret.Data[keyName]
+
+	if !ok {
+		return "", errors.New("key was not in secret data")
+	}
+
+	return string(realmKey), nil
+}
+
+func GetRealmKeyArgs(context *clusterd.Context, realmName string, namespace string) (string, string, error) {
+	// get realm's access and secret keys
+	realmSecretName := realmName + "-keys"
+	realmSecret, err := context.Clientset.CoreV1().Secrets(namespace).Get(realmSecretName, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return "", "", errors.Wrapf(err, "failed to get realm %q keys secret", realmName)
+		}
+	}
```

Comment:
Don't we need to return also if the secret is not found? Otherwise `realmSecret` will be nil and this method will later fail, right?